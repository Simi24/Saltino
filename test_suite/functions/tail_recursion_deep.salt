// Test di deep tail recursion - questo test dovrebbe funzionare solo con TCO
def main() {
    return deep_recursion_tail(1000, 0)
}

def deep_recursion_tail(n, acc) {
    if (n <= 0) {
        return acc
    } else {
        return deep_recursion_tail(n - 1, acc + 1)
    }
}
