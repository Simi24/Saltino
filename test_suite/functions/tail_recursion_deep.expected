Program result: 1000
